amazonsqs/tests/test.bal

```diff
@@ -0,0 +1,108 @@
+// Copyright (c) 2018 WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+//
+// WSO2 Inc. licenses this file to you under the Apache License,
+// Version 2.0 (the "License"); you may not use this file except
+// in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+import ballerina/config;
+import ballerina/test;
+import ballerina/io;
+import ballerina/log;
+import ballerina/encoding;
+
+Configuration configuration = {
+    accessKey: config:getAsString("ACCESS_KEY_ID"),
+    secretKey: config:getAsString("SECRET_ACCESS_KEY"),
+    region: config:getAsString("REGION"),
+    accountNumber: config:getAsString("ACCOUNT_NUMBER")
+};
+
+Client sqsClient = new(configuration);
+
+@test:Config
+function testCreateQueue() {
+
+    map<string> attributes = {};
+    attributes["VisibilityTimeout"] = "400";
+    attributes["FifoQueue"] = "true";
+
+    string|error response = sqsClient->createQueue("demox.fifo", attributes);
+    if(response is string && response.hasPrefix("http")) {
+        log:printInfo("Created queue: \n" + response);
+        test:assertTrue(true);
+    } else {
+        test:assertTrue(false);
+    }
+}
+
+@test:Config
+function testSendMessage() {
+
+    map<string> attributes = {};
+    attributes["MessageDeduplicationId"] = "dupID2";
+    attributes["MessageGroupId"] = "grpID1";
+    attributes["MessageAttribute.1.Name"] = "N1";
+    attributes["MessageAttribute.1.Value.StringValue"] = "V1";
+    attributes["MessageAttribute.1.Value.DataType"] = "String";
+    attributes["MessageAttribute.2.Name"] = "N2";
+    attributes["MessageAttribute.2.Value.StringValue"] = "V2";
+    attributes["MessageAttribute.2.Value.DataType"] = "String";
+
+    string queueUrl = "";
+    OutboundMessage|error response = sqsClient->sendMessage("New Message Text", "/610968236798/demox.fifo", attributes);
+    if(response is OutboundMessage) {
+        log:printInfo("Response from SQS: \n");
+        io:println(response);
+        test:assertTrue(true);
+    } else {
+        test:assertTrue(false);
+    }
+}
+
+@test:Config
+function testReceiveMessage() {
+
+    map<string> attributes = {};
+    attributes["MaxNumberOfMessages"] = "1";
+    attributes["VisibilityTimeout"] = "600";
+    attributes["WaitTimeSeconds"] = "2";
+    attributes["AttributeName.1"] = "SenderId";
+    attributes["MessageAttributeName.1"] = "N2";
+
+    InboundMessage[]|error response = sqsClient->receiveMessage("/610968236798/demox.fifo", attributes);
+    if(response is InboundMessage[]) {
+        log:printInfo("Received from SQS: \n");
+        io:println(response);
```

Comment:
Can't we use log:printInfo() to log the response? Otherwise it will be missed in the log file.

Reply:
Corrected